benchmarks/domains/synthetic_2C1D_1C.py

```diff
@@ -0,0 +1,131 @@
+from benchmarks.definition import BenchmarkDefinition
+from benchmarks.definition.config import (
+    BenchmarkFunctionDefinition,
+    ConvergenceExperimentSettings,
+)
+
+if TYPE_CHECKING:
+    from mpl_toolkits.mplot3d import Axes3D
+
+
+def lookup(z: np.ndarray, x: np.ndarray, y: np.ndarray) -> np.ndarray:
+    """Lookup function.
+
+    Inputs:
+        z   discrete   {1,2,3,4}
+        x   continuous [-2*pi, 2*pi]
+        y   continuous [-2*pi, 2*pi]
+    Output: continuous
+    Objective: Maximization
+    Optimal Inputs:
+        {x: 1.610, y: 1.571, z: 3}
+        {x: 1.610, y: -4.712, z: 3}
+    Optimal Output: 4.09685
+    """
+    try:
+        assert np.all(-2 * pi <= x) and np.all(x <= 2 * pi)
+        assert np.all(-2 * pi <= y) and np.all(y <= 2 * pi)
+        assert np.all(np.isin(z, [1, 2, 3, 4]))
+    except AssertionError:
+        raise ValueError("Inputs are not in the valid ranges.")
+
+    return (
+        (z == 1) * sin(x) * (1 + sin(y))
+        + (z == 2) * (x * sin(0.9 * x) + sin(x) * sin(y))
+        + (z == 3) * (sqrt(x + 8) * sin(x) + sin(x) * sin(y))
+        + (z == 4) * (x * sin(1.666 * sqrt(x + 8)) + sin(x) * sin(y))
+    )
+
+
+def benchmark_callable(scenario_config: ConvergenceExperimentSettings) -> DataFrame:
+    """Optimization benchmark with two continuous and one discrete input."""
+    parameters = [
+        NumericalContinuousParameter("x", (-2 * pi, 2 * pi)),
+        NumericalContinuousParameter("y", (-2 * pi, 2 * pi)),
+        NumericalDiscreteParameter("z", (1, 2, 3, 4)),
+    ]
+
+    objective = NumericalTarget(name="target", mode=TargetMode.MAX).to_objective()
+    search_space = SearchSpace.from_product(parameters=parameters)
+
+    scenarios: dict[str, Campaign] = {
+        "Random Recommender": Campaign(
+            searchspace=search_space,
+            recommender=RandomRecommender(),
+            objective=objective,
+        ),
+        "Default Recommender": Campaign(
+            searchspace=search_space,
+            objective=objective,
+        ),
+    }
+
+    return simulate_scenarios(
+        scenarios,
+        lookup,
+        batch_size=scenario_config.batch_size,
+        n_doe_iterations=scenario_config.n_doe_iterations,
+        n_mc_iterations=scenario_config.n_mc_iterations,
+        impute_mode="error",
+        random_seed=scenario_config.random_seed,
+    )
+
+
+benchmark_config = ConvergenceExperimentSettings(
+    batch_size=5,
+    n_doe_iterations=30,
+    n_mc_iterations=50,
+)
+
+description = ""
+if lookup.__doc__:
+    description = lookup.__doc__
+
+benchmark_function_definition = BenchmarkFunctionDefinition(
+    callable=benchmark_callable,
+    description=description,
+    best_possible_result=4.09685,
+)
+
+benchmark = BenchmarkDefinition(
+    identifier="synthetic_2C1D_1C",
+    benchmark_function_definition=benchmark_function_definition,
+    settings=benchmark_config,
+)
+
+
+if __name__ == "__main__":
```

Comment:
Why do we now suddenly have code related to visualization here?

Reply:
Well, we explicitly discussed that we do not want to have things like that here if I remember. SO unlesss @AdrianSosic manages to convinve me, I'd say this should go.

Reply:
Just pasting the answer from our chat here, for the protocol:

I think it makes sense to have some visualization available where ever this is possible. Here triggered when running the domain file as a script made most sense to me, but this is something that we can always change without any implications later. NOT having a a way to visualize make it really hard to understand what these benchmarks are about. In fact, the reason why I added it was when I had to figure out what are the max and argmax values, which would have been difficult without the visualization